Fix #21372 - Admin Search Suggestions Toggle Not Working

Stoyvo opened the pull request:
Fix reported bug of broken admin toggle for frontend

### Description (*)
Loads system config value for Enable/Disable search suggestions, add conditional check on search suggestion request.

### Fixed Issues (if relevant)
1. magento/magento2#21372: Admin Search Suggestions Toggle Not Working

### Manual testing scenarios (*)
Scenario 1:
1. Enable Search Suggestions in Admin
2. Enter a search result on frontend (submit it, press enter)
3. Begin entering the same search query
4. Suggestions should appear

Scenario 2:
1. Disable Search Suggestions in Admin
2. Enter a search result on frontend (submit it, press enter)
3. Begin entering the same search query
4. Suggestions should **NOT** appear


### Contribution checklist (*)
 - [ x ] Pull request has a meaningful description of its purpose
 - [ x ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Stoyvo committed:
Search FormMini Block referenced in layout

Stoyvo committed:
Create FormMini Block, Serialize config value

Stoyvo committed:
Add conditional check to execute search suggestions

magento-engcom-team commented:
Hi @Stoyvo. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

Stoyvo committed:
Removed unused classes

Stoyvo committed:
Add JS searchConfig serialized object to window

Stoyvo committed:
Update FormMini.php

rogyar committed:
Minor refactoring

Stoyvo commented on app/code/Magento/Search/Block/FormMini.php:
```diff
@@ -3,22 +3,21 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
```
@rogyar What does this strict_types do?

Stoyvo committed:
Fix Go Button

Go button is disabled=true by default. If search suggestions are disabled, we should not disable this button.

rogyar reviewed (changes requested):
Hi @Stoyvo. Could you please check the failing tests

sidolov commented:
@Stoyvo , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!
